### PR TITLE
Convert Tuples to Records and split a helper out from findCanonicalModelElementFor

### PR DESCRIPTION
**Pull request by srawlins**

### CODE REVIEW

**Comment by parlough** [comment]
I've been looking forward to that tuple file disappearing for a long time 🥳

**Review by jcollins-g — APPROVED** [review]
Nice!

File: lib/src/tuple.dart [hunk 1]
```diff
@@ -1,56 +0,0 @@
-// Copied from source at github.com/kseo/tuple/blob/470ed3aeb/lib/src/tuple.dart
```
Comment (kevmoo) [inline lib/src/tuple.dart:1]: 🎉🎉🎉

Comment (jcollins-g) [reply lib/src/tuple.dart:1]: > 🎉🎉🎉

Haha!

**Pull request merged**